src/Ansatz/Chain.jl

```diff
@@ -30,52 +30,110 @@ function Chain(tn::TensorNetwork, sites, args...; kwargs...)
     Chain(Quantum(tn, sites), args...; kwargs...)
 end
 
-function Chain(::State, boundary::Periodic, arrays::Vector{<:AbstractArray})
+defaultorder(::State) = (:o, :l, :r)
+defaultorder(::Operator) = (:o, :i, :l, :r)
+
+function Chain(::State, boundary::Periodic, arrays::Vector{<:AbstractArray}; order = defaultorder(State()))
     @assert all(==(3) ∘ ndims, arrays) "All arrays must have 3 dimensions"
+    issetequal(order, defaultorder(State())) ||
+        throw(ArgumentError("order must be a permutation of $(String.(defaultorder(State())))"))
 
     n = length(arrays)
     symbols = [nextindex() for _ in 1:2n]
 
+    function get_index(directions, i)
+        map(directions) do dir
+            if dir == :o
+                symbols[i]
+            elseif dir == :r
+                symbols[n+mod1(i, n)]
+            elseif dir == :l
+                symbols[n+mod1(i - 1, n)]
+            else
+                throw(ArgumentError("Invalid direction: $dir"))
+            end
+        end
+    end
```

Comment:
You're only using this code once. Remove this function and move the code inside the place where you are using it.

Reply:
Okey